src/LndClient/RPC/Katip.hs

```diff
@@ -127,3 +130,29 @@ ensureHodlInvoice env req =
               AddInvoice.paymentRequest = Invoice.paymentRequest x,
               AddInvoice.addIndex = Invoice.addIndex x
             }
+
+closeChannelSync ::
+  (KatipContext m, MonadUnliftIO m) =>
+  LndEnv ->
+  CloseChannelRequest ->
+  m (Either LndError ())
+closeChannelSync = closeChannelRecursive 10
+  where
+    closeChannelRecursive (0 :: Int) env _ = do
+      $(logTM)
+        (newSeverity env ErrorS Nothing Nothing)
+        "Channel couldn't be closed."
+      return $ Left $ LndError "Cannot close channel"
+    closeChannelRecursive n env req = do
+      mVar <- newEmptyMVar
```

Comment:
It's more efficient to create MVar outside of the recursive function. In case where 1 sec delay was not enough, but channel status was switched to pending in let's say 1.5 sec - it can help to avoid unneeded retries, because all already spawned processes will be able to put value into this MVar

Reply:
agree, done, thanks